src/validator/CreditCardValidator.java

```
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
```

Suggesting to redesign this class to a service:

- Rename this class to CreditCardValidatorService, drop the redundant interface as stated earlier.
- Add and inject credit card vendor name to credit card verification map ex: 'private final Map<String, CreditCardVendor> vendors;'
- The service will have a single public responsibility: 'public boolean public boolean checkValidity(String vendorName, String number) throws IllegalArgumentException'. (Should throw on unknown vendor name)